plugins/text-editor-resources/src/components/EmojiPopup.svelte

```diff
@@ -0,0 +1,134 @@
+  import { Label, ListView, resizeObserver } from '@hcengineering/ui'
+  import { createEventDispatcher } from 'svelte'
+  import emojiReplaceDict from './extension/emojiIdMap.json'
+
+  export let query: string = ''
+
+  let items = Object.entries(emojiReplaceDict)
+
+  const dispatch = createEventDispatcher()
+
+  let list: ListView
+  let scrollContainer: HTMLElement
+  let selection = 0
+
+  function dispatchItem(item: [string, string]): void {
+    dispatch('close', {
+      id: item[0],
+      objectclass: item[1]
+    })
+  }
+
+  export function onKeyDown(key: KeyboardEvent): boolean {
+    if (key.key === 'ArrowDown') {
+      key.stopPropagation()
+      key.preventDefault()
+      list?.select(selection + 1)
+      return true
+    }
+    if (key.key === 'ArrowUp') {
+      key.stopPropagation()
+      key.preventDefault()
+      if (selection === 0 && scrollContainer !== undefined) {
+        scrollContainer.scrollTop = 0
+      }
+      list?.select(selection - 1)
+      return true
+    }
+    if (key.key === 'Enter' || key.key === 'Tab') {
+      key.preventDefault()
+      key.stopPropagation()
+      if (selection < items.length) {
+        const searchItem = items[selection]
+        dispatchItem(searchItem)
+        return true
+      } else {
+        return false
+      }
+    }
+    return false
+  }
+
+  const updateItems = reduceCalls(async function (localQuery: string): Promise<void> {
+    items = Object.entries(emojiReplaceDict).filter(([k, v]) => k.includes(localQuery))
+  })
+  $: void updateItems(query)
+</script>
+
+{#if (items.length === 0 && query !== '') || items.length > 0}
+  <!-- svelte-ignore a11y-no-noninteractive-element-interactions -->
+  <form class="antiPopup mentionPoup" on:keydown={onKeyDown} use:resizeObserver={() => dispatch('changeSize')}>
+    <div class="ap-scroll" bind:this={scrollContainer}>
+      <div class="ap-box">
+        {#if items.length === 0 && query !== ''}
+          <div class="noResults"><Label label={presentation.string.NoResults} /></div>
+        {/if}
+        {#if items.length > 0}
+          <!-- svelte-ignore a11y-click-events-have-key-events -->
+          <ListView bind:this={list} bind:selection count={items.length}>
+            <svelte:fragment slot="item" let:item={num}>
+              {@const item = items[num]}
+              <!-- svelte-ignore a11y-no-static-element-interactions -->
+              <div
+                class="ap-menuItem withComp h-8"
+                on:click={() => {
+                  dispatchItem(item)
+                }}
+              >
+                <div class="flex-row-center">
+                  <div class="flex-center p-1 content-dark-color flex-no-shrink">
+                    {item[1]}
+                  </div>
+                  <span class="ml-1 max-w-120 overflow-label searchResult">
+                    <span class="name">{item[0]}</span>
+                  </span>
+                </div>
+              </div>
```

Comment:
Semantic html please!

I cannot help you with that here since I am not fully aware of the context. But the least we could do for accessibility is to add `aria-*` or other html attributes.

Reply:
I did how they did. 
https://github.com/ElSalvo96/platform/blob/4e7673cb5b1af10e66cb84f05fba3790494b05d4/packages/presentation/src/components/SearchResult.svelte#L27-L46

What kind of improvement do you expect with the `aria-*` tag? I think there you don't need any of these